bson/src/test/unit/util/JsonPoweredTestHelper.java

```diff
@@ -87,7 +87,15 @@ public static Map<Path, BsonDocument> testDocs(final Path dir) {
 
     public static List<File> getTestFiles(final String resourcePath) throws URISyntaxException {
         List<File> files = new ArrayList<>();
-        addFilesFromDirectory(new File(JsonPoweredTestHelper.class.getResource(resourcePath).toURI()), files);
+
+        URL resource = JsonPoweredTestHelper.class.getResource(resourcePath);
+        File directory;
+        try {
+            directory = new File(resource.toURI());
+        } catch (IllegalArgumentException e) {
+            directory = new File(resource.toExternalForm());
```

Comment:
Could we use `File.exists` instead of catching an exception and retrying?

Reply:
Did an alternative check - looking for jar in the path